--- a/arch/mips/ath79/Kconfig.openwrt
+++ b/arch/mips/ath79/Kconfig.openwrt
@@ -1960,6 +1960,16 @@ config ATH79_MACH_TL_WR941ND_V6
 	select ATH79_DEV_M25P80
 	select ATH79_DEV_WMAC
 
+config ATH79_MACH_TL_WR941N_V7
+	bool "TP-LINK TL-WR941N v7 support"
+	select SOC_QCA955X
+	select ATH79_DEV_ETH
+	select ATH79_DEV_GPIO_BUTTONS
+	select ATH79_DEV_LEDS_GPIO
+	select ATH79_DEV_M25P80
+	select ATH79_DEV_USB
+	select ATH79_DEV_WMAC
+
 config ATH79_MACH_TL_WR940N_V4
 	bool "TP-LINK TL-WR940N v4 support"
 	select SOC_QCA956X
--- a/arch/mips/ath79/Makefile
+++ b/arch/mips/ath79/Makefile
@@ -247,6 +247,7 @@ obj-$(CONFIG_ATH79_MACH_TL_WR841N_V9)		+
 obj-$(CONFIG_ATH79_MACH_TL_WR902AC_V1)		+= mach-tl-wr902ac-v1.o
 obj-$(CONFIG_ATH79_MACH_TL_WR941ND)		+= mach-tl-wr941nd.o
 obj-$(CONFIG_ATH79_MACH_TL_WR941ND_V6)		+= mach-tl-wr941nd-v6.o
+obj-$(CONFIG_ATH79_MACH_TL_WR941N_V7)		+= mach-tl-wr941n-v7.o
 obj-$(CONFIG_ATH79_MACH_TL_WR940N_V4)		+= mach-tl-wr940n-v4.o
 obj-$(CONFIG_ATH79_MACH_TL_WR942N_V1)		+= mach-tl-wr942n-v1.o
 obj-$(CONFIG_ATH79_MACH_TS_D084)		+= mach-ts-d084.o
--- a/arch/mips/ath79/machtypes.h
+++ b/arch/mips/ath79/machtypes.h
@@ -306,6 +306,7 @@ enum ath79_mach_type {
 	ATH79_MACH_TL_WR941ND,			/* TP-LINK TL-WR941ND */
 	ATH79_MACH_TL_WR941ND_V5,		/* TP-LINK TL-WR941ND v5 */
 	ATH79_MACH_TL_WR941ND_V6,		/* TP-LINK TL-WR941ND v6 */
+	ATH79_MACH_TL_WR941N_V7,		/* TP-LINK TL-WR941N v7 */
 	ATH79_MACH_TL_WR940N_V4,		/* TP-LINK TL-WR940N v4 */
 	ATH79_MACH_TL_WR940N_V6,		/* TP-LINK TL-WR940N v6 */
 	ATH79_MACH_TL_WR942N_V1,		/* TP-LINK TL-WR942N v1 */
